atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SpecificTableSweeper.java

```diff
@@ -126,15 +126,21 @@ void runOnceAndSaveResults(TableToSweep tableToSweep, SweepBatchConfig batchConf
 
     SweepResults runOneIteration(
             TableReference tableRef, byte[] startRow, SweepBatchConfig batchConfig, SweepTaskRunner.RunType runType) {
-        try {
```

Comment:
Could we have a finite number of retries, maybe 50 or something?

The problem here is that there is an edge case where if the network link between an Atlas node A1 and a Cassandra node C1 fails even though there exists another Atlas node A2 that can talk to C1, then we deadlock (because this is run holding the sweep locks) and this could starve background sweep.